lib/_tls_common.js

```diff
@@ -153,6 +153,31 @@ exports.createSecureContext = function createSecureContext(options) {
     }
   }
 
+  const { privateKeyIdentifier, privateKeyEngine } = options;
+  if (privateKeyIdentifier !== undefined && privateKeyEngine !== undefined) {
```

Comment:
Was it your intention that specifying a `privateKeyIdentifier` while forgetting to set the `privateKeyEngine` would be silently ignored? I can see an argument for ignoring the engine if no key identifier is specified, but I'm having more trouble seeing the case for ignoring the key if the engine is not specified.

Reply:
Yes, that was the idea, but I can throw if that is better.